Adapt vscode-r-test-adapter

jennybc opened the pull request:
Addresses #1365

jennybc committed:
Adapt vscode-r-test-adapter

Adapt https://github.com/meakbiyik/vscode-r-test-adapter for use in the positron R extension

jennybc committed:
Correct this comment

split2 is new, child_process is not

jennybc committed:
Add license info about vscode-r-test-adapter

jennybc committed:
Get rid of one layer of helper function

DavisVaughan reviewed (approved):
I think in general this feels like a really great first step!

Lots of comments, nothing that should block this MVP from being merged. The most important one right now is that it is currently dramatically slower to run a full test suite this way.

Otherwise the comments are mostly about workflow. We should turn important comments / discussions in this thread into issues so we can track them

jennybc committed:
Turn on testing support by default

jennybc committed:
Support on-the-fly configuration for R test explorer

jennybc committed:
Simplify the file watchers

It seems weird to have "special case worries" about files matching this pattern. Until I see the need with my own eyes, let's live without this.

jennybc committed:
Be more responsive to current testthat state

jennybc committed:
Fill in authors for the MIT license
